src/psyclone/tests/domain/common/algorithm/alg_psyir_test.py

```diff
@@ -553,24 +551,33 @@ def test_aic_lowertolanguagelevel_multi_invoke():
     assert isinstance(invoke2, AlgorithmInvokeCall)
     assert len(psyir.walk(AlgorithmInvokeCall)) == 2
     assert len(psyir.walk(KernelFunctor)) == 2
-    assert invoke1.scope.symbol_table.lookup("invoke")
-    assert invoke2.scope.symbol_table.lookup("invoke")
+    assert "invoke" in invoke1.scope.symbol_table._symbols
+    assert "invoke" in invoke2.scope.symbol_table._symbols
 
     # Don't call create_psylayer_symbol_root_names() here. This is to
     # check that lower_to_language_level() creates the names if
     # needed.
-    psyir.lower_to_language_level()
 
-    assert len(psyir.walk(AlgorithmInvokeCall)) == 0
-    assert len(psyir.walk(KernelFunctor)) == 0
+    # Just lower one of the invoke's. The 'invoke' symbol should still
+    # exist.
+    invoke1.lower_to_language_level()
     call1 = psyir.children[0][0]
-    with pytest.raises(KeyError) as info:
-        call1.scope.symbol_table.lookup("invoke")
-    assert "Could not find 'invoke' in the Symbol Table." in str(info.value)
+    assert not isinstance(call1, AlgorithmInvokeCall)
+    assert isinstance(invoke2, AlgorithmInvokeCall)
+    assert len(psyir.walk(AlgorithmInvokeCall)) == 1
+    assert len(psyir.walk(KernelFunctor)) == 1
+    assert "invoke" in call1.scope.symbol_table._symbols
+    assert "invoke" in invoke2.scope.symbol_table._symbols
+
+    # Now lower the second invoke. The 'invoke' symbol should be
+    # removed.
+    invoke2.lower_to_language_level()
     call2 = psyir.children[0][1]
-    with pytest.raises(KeyError) as info:
-        call2.scope.symbol_table.lookup("invoke")
-    assert "Could not find 'invoke' in the Symbol Table." in str(info.value)
+
+    assert len(psyir.walk(AlgorithmInvokeCall)) == 0
```

Comment:
Probably should be `assert not psyir.walk(AlgorithmInvokeCall)` and again below.

Reply:
Updated all cases in the file.